sway-core/src/type_system/substitute/subst_types.rs

```diff
@@ -1,11 +1,52 @@
 use crate::{engine_threading::*, type_system::priv_prelude::*};
 
 pub trait SubstTypes {
-    fn subst_inner(&mut self, type_mapping: &TypeSubstMap, engines: &Engines);
+    fn subst_inner(&mut self, type_mapping: &TypeSubstMap, engines: &Engines) -> bool;
 
-    fn subst(&mut self, type_mapping: &TypeSubstMap, engines: &Engines) {
+    fn subst(&mut self, type_mapping: &TypeSubstMap, engines: &Engines) -> bool {
```

Comment:
These need documentation of what the `bool` result means at the very least.
Perhaps better would be some idiomatic enum.

Reply:
Done.